src/inmanta/data/__init__.py

```diff
@@ -1338,11 +1382,35 @@ async def get_by_id(
             return result[0]
         return None
 
+    @classmethod
+    async def lock_by_id(
```

Comment:
I added this and the row locking functionality because the initial implementation used it. This is no longer the case. I assume I'd best remove this method, the `RowLockMode` class and all its uses to avoid code rot?